src/genesis/export_to_csv.py

```diff
@@ -0,0 +1,252 @@
+        print(response.text)
+        return None
+
+def get_analysis(graphql_query, food_id):
+    nutrients = {}
+
+    variables = {
+        "input":{
+            "foodId": food_id,
+            "analysisInput":{
+                "analysisType": "Net",
+                "amount":{
+                    "quantity":"100",
+                    "unitId":"a7df0af5-0008-0000-7484-751e8eaf05c6"
+                }
+            }
+        }
+    }
+
+    result = run_query(graphql_query, variables)
+    if result:
+        nutrients = result.get("data", {}).get("analysis", {}).get("getAnalysis", {}).get("analysis", {}).get("nutrientInfos",[])
+        if len(nutrients) == 0:
+            print(f"Unable to get nutrient information for {food_id}")
+
+    return nutrients
+
+def search(graphql_query, food_type):
+    result_items = []
+    variables = {
+        "input": {
+            "searchText": '*',
+            "foodTypes": [food_type],
+            "itemSourceFilter": "Customer",
+            "archiveFilter" : "Unarchived",
+            "versionFilter": "Latest",
+            "first": output_limit,
+            "after": 0
+        }
+    }
+
+    print(f"Running query...")
+    result = run_query(graphql_query, variables)
+    if result:
+        total_count = result.get("data", {}).get("foods", {}).get("search", {}).get("totalCount", 0)
+        print(f"Found {total_count} results.")
+
+    return result
+
+def export_to_json(result):
+    first_entry = True
+    returned_items = []
+
+    """Iterate through each character in the corpus and update the searchText."""
+    with open(file_path, "a") as file:
+
+        file.write("[\n")  # Write the opening bracket for the JSON array
+        if not first_entry:
+            file.write(",\n")  # Add a comma before each entry - except the first
+
+        json.dump(result, file, indent=4)  # Append the JSON response to the file
+        first_entry = False
+
+        file.write("\n]")  # Write the closing bracket for the JSON array
+
+def json_to_csv(food_results, csv_file):
+    """ Convert a JSON array to CSV format.
+        Expectation that this will be the foodSearchResult element of the array
+    """
+
+    # Delete the file if it already exists
+    if os.path.exists(csv_file):
+        os.remove(csv_file)
+        print(f"Existing file '{csv_file}' has been deleted.")
+
+    # Different results may have different fields; handle
+    fieldnames = []
+    for r in food_results:
+        for key in r.keys():
+            if key not in fieldnames:
+                fieldnames.append(key)
+
+    # Write JSON data to CSV
+    with open(csv_file, mode='w', newline='') as file:
+        # Use the keys of the first item as the header
+        #fieldnames = food_results[0].keys()
+        writer = csv.DictWriter(file, fieldnames=fieldnames)
+
+        # Write header and rows
+        writer.writeheader()
+        writer.writerows(food_results)
+
+
+def json_file_to_csv(json_file, csv_file):
```

Comment:
I don't see this being used anywhere. It should be removed for clarity.

Reply:
Removed